solc/CommandLineInterface.cpp

```diff
@@ -948,6 +950,7 @@ bool CommandLineInterface::processInput()
 		Input inputLanguage = m_args.count(g_argYul) ? Input::Yul : (m_args.count(g_argStrictAssembly) ? Input::StrictAssembly : Input::Assembly);
 		Machine targetMachine = Machine::EVM;
 		bool optimize = m_args.count(g_argOptimize) || m_args.count(g_strOptimizeYul);
+		// TODO: should we exclude --no-optimize-yul here? Or enable optimizing by default?
```

Comment:
In assembly mode, you should use ``--optimize`` instead of ``--optimize-yul``, so I think it can be ignored.